modules/openapi-generator/src/main/resources/python/python-experimental/signing.mustache

```diff
@@ -0,0 +1,317 @@
+
+        request_headers_dict[HEADER_AUTHORIZATION] = self.get_authorization_header(
+            signed_headers_dict, b64_signed_msg)
+
+        return request_headers_dict
+
+    def load_private_key(self):
+        """Load the private key used to sign HTTP requests.
+            The private key is used to sign HTTP requests as defined in
+            https://datatracker.ietf.org/doc/draft-cavage-http-signatures/.
+        """
+        if self.private_key is not None:
+            return
+        with open(self.private_key_path, 'rb') as f:
+            # Decode PEM file and determine key type from PEM header.
+            # Supported values are "RSA PRIVATE KEY" and "ECDSA PRIVATE KEY".
+            keys = pem.parse(f.read())
+            if len(keys) != 1:
+                raise Exception("File must contain exactly one private key")
+            key = keys[0].as_text()
+            if key.startswith('-----BEGIN RSA PRIVATE KEY-----'):
+                self.private_key = RSA.importKey(key)
+            elif key.startswith('-----BEGIN EC PRIVATE KEY-----'):
+                self.private_key = ECC.importKey(key)
+            else:
+                raise Exception("Unsupported key")
+
+    def get_unix_time(self, ts):
+        """Converts and returns a datetime object to UNIX time, the number of seconds
+           elapsed since January 1, 1970 UTC.
+        """
+        return (ts - datetime.datetime(1970,1,1)).total_seconds())
+
+    def get_signed_header_info(self, host, resource_path, method, headers, body, query_params):
+        """
+        Build the HTTP headers (name, value) that need to be included in the HTTP signature scheme.
+
+        :param resource_path : A string representation of the HTTP request resource path.
+        :param method: A string representation of the HTTP request method, e.g. GET, POST.
+        :param headers: A dict containing the HTTP request headers.
+        :param body: The string representation of the HTTP request body.
+        :param query_params: A string representing the HTTP request query parameters.
+        :return: A tuple containing two dict objects:
+            The first dict contains the HTTP headers that are used to calculate the HTTP signature.
+            The second dict contains the HTTP headers that must be added to the outbound HTTP request.
+        """
+
+        if body is None:
+            body = ''
+        else:
+            body = json.dumps(body)
+
+        # Build the '(request-target)' HTTP signature parameter.
+        target_host = urlparse(host).netloc
+        target_path = urlparse(host).path
+        request_target = method.lower() + " " + target_path + resource_path
+        if query_params:
+            raw_query = urlencode(query_params).replace('+', '%20')
+            request_target += "?" + raw_query
+
+        # Get current time and generate RFC 1123 (HTTP/1.1) date/time string.
+        now = datetime.datetime.now()
```

Comment:
Now changes depending upon what time zone the machine is in running this code.
Can you add a constant time_zone which defaults to tzutc in this class's `__init__` method?
That way our user can specify what timezone now is calculated in.
Can you update now an unix time zero to use that timezone?
See an example of timezone usage here: https://medium.com/@chaoren/datetime-to-unix-time-epoch-in-python-2-3-ecba6493e8b

Reply:
The link you provided uses strftime('%s'), but it's a bad idea to use that. See
https://stackoverflow.com/questions/11743019/convert-python-datetime-to-epoch-with-strftime

 I was reading even though there are many references to , it's a bad idea to use it

Reply:
Good to know, thanks.

Reply:
The code is essentially calculating the deltatime (now - 01/01/1970), where both now and 01/01/1970 are datetime objects are in the same timezone. It may be UTC, PST or any time zone, but both arguments are in the same timezone, so shouldn't the outcome be exactly the same?

Reply:
> Good to know, thanks. Sorry for that.

I know. It's everywhere, initially I also used strftime('%s'). There is a better option but unfortunately it only works for Python 3.3 and above.

Reply:
Marking resolved for now.